lib/broccoli/ember-app.js

```diff
@@ -1097,14 +1138,19 @@ class EmberApp {
         overwrite: true,
       });
 
-      if (this.amdModuleNames) {
-        let anonymousAmd = new Funnel(externalTree, {
-          files: Object.keys(this.amdModuleNames),
-          annotation: 'Funnel (named AMD)',
-        });
-        externalTree = this._mergeTrees([externalTree, shimAmd(anonymousAmd, this.amdModuleNames)], {
-          annotation: 'TreeMerger (named AMD)',
-          overwrite: true,
+      if (this.customTransformOrder) {
+        this.customTransformOrder.forEach(transformName => {
+          let transformConfig = this.customTransforms[transformName];
+
+          let transformTree = new Funnel(externalTree, {
+            files: transformConfig.files,
+            annotation: `Funnel (custom transform: ${transformName})`,
+          });
+
+          externalTree = this._mergeTrees([externalTree, transformConfig.callbackFn(transformTree, transformConfig.options)], {
```

Comment:
Yes, `transformConfig.options` is a map of asset path to the options provided by `app.import`. We pass the whole map and read off the assetPath